astropy/coordinates/representation.py

```diff
@@ -230,6 +234,140 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def _scale_operation(self, op, *args):
+        results = []
+        for component, cls in self.attr_classes.items():
+            value = getattr(self, component)
+            if issubclass(cls, Angle):
+                results.append(value)
+            else:
+                results.append(op(value, *args))
+
+        # try/except catches anything that cannot initialize the class, such
+        # as operations that returned NotImplemented or a representation
+        # instead of a quantity (as would happen for, e.g., rep * rep).
+        try:
+            return self.__class__(*results)
+        except Exception:
+            return NotImplemented
+
+    def __mul__(self, other):
+        return self._scale_operation(operator.mul, other)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        return self._scale_operation(operator.truediv, other)
+
+    def __div__(self, other):
+        return self._scale_operation(operator.truediv, other)
+
+    def __neg__(self):
+        return self._scale_operation(operator.neg)
+
+    # Follow numpy convention and make an independent copy.
+    def __pos__(self):
+        return self.__class__(*(getattr(self, component)
+                                for component in self.components), copy=True)
+
+    def __abs__(self):
+        return np.sqrt(functools.reduce(
+            operator.add, (getattr(self, component)**2
```

Comment:
I think we should avoid conversions to ```Distance``` - one might want to use ```CartesianRepresentations``` to represent velocities for example.